### PR TITLE
Pass rustc location via RUSTC when invoking cargo

### PR DESCRIPTION
**Pull request by madisp**

### CODE REVIEW

**Comment by matklad** [comment]
Will this potentially break installations where `rustc` and `cargo` are not in the same directory, but `rustc` is in `PATH`? Are there any such layouts? @jajakobyly is it correct that on windows `rustc` and `cargo` reside in the same directory?

File: src/main/kotlin/org/rust/cargo/commands/Cargo.kt [hunk 1]
```diff
@@ -53,6 +55,7 @@ class Cargo(
             .withParameters(command)
             .withParameters(additionalArguments)
             .withEnvironment(environmentVariables)
+            .withEnvironment(CargoConstants.RUSTC_ENV_VAR, pathToRustExecutable)
```
Comment (matklad) [inline src/main/kotlin/org/rust/cargo/commands/Cargo.kt]: I think this should come before `withEnvironment(environmentVariables)` to be able to override `RUSTC`.

Comment (madisp) [inline src/main/kotlin/org/rust/cargo/commands/Cargo.kt]: Doesn't really matter, `withEnvironment(Map)` does a `putAll()` and `withEnvironment(String, String)` does a put.

Comment (madisp) [inline src/main/kotlin/org/rust/cargo/commands/Cargo.kt]: Will change, how do you usually do this, with a separate commit or a force push to the branch?

Comment (matklad) [inline src/main/kotlin/org/rust/cargo/commands/Cargo.kt]: Does not matter much, this is one commit length pull request, so "squash merge" would work anyway.

**Comment by matklad** [comment]
I think that this should not break anything (will check windows story), but ideally we should provide an *optional* "rustc path" settings in the GUI. And teaching `cargo` not to fail if there is no `rustc` is a also good idea: https://github.com/rust-lang/cargo/issues/2833

**Comment by matklad** [comment]
Thanks!

**Comment by jajakobyly** [comment]
> @jajakobyly is it correct that on windows rustc and cargo reside in the same directory? 

yup

**Pull request merged**